src/commands/executeFunction/executeEventGridFunction.ts

```diff
@@ -0,0 +1,156 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type IActionContext, type IAzureQuickPickItem } from '@microsoft/vscode-azext-utils';
+import * as fs from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { localize } from '../../localize';
+import { type FunctionTreeItemBase } from '../../tree/FunctionTreeItemBase';
+import { feedUtils } from '../../utils/feedUtils';
+import { type IFunction } from '../../workspace/LocalFunction';
+import { supportedEventGridSourceLabels, supportedEventGridSources, type EventGridSource } from './eventGridSources';
+import { executeFunctionWithInput } from './executeFunction';
+
+type FileMetadata = {
+    name: string;
+    path: string;
+    sha: string;
+    size: number;
+    url: string;
+    html_url: string;
+    git_url: string;
+    download_url: string;
+    type: string;
+    _links: {
+        self: string;
+        git: string;
+        html: string;
+    };
+};
+
+const sampleFilesUrl =
+    'https://api.github.com/repos/Azure/azure-rest-api-specs/contents/specification/eventgrid/data-plane/' +
+    '{eventSource}' +
+    '/stable/2018-01-01/examples/cloud-events-schema/';
+
+export async function executeEventGridFunction(context: IActionContext, node: FunctionTreeItemBase | IFunction): Promise<void> {
+    // Prompt for event source
+    const eventGridSourcePicks: IAzureQuickPickItem<EventGridSource | undefined>[] = supportedEventGridSources.map((source: EventGridSource) => {
+        return {
+            label: supportedEventGridSourceLabels.get(source) || source,
+            data: source,
+        };
+    });
+    const eventSource: EventGridSource =
+        (
+            await context.ui.showQuickPick(eventGridSourcePicks, {
```

Comment:
All of these quickPick calls should be broken up into separate wizard steps. This gives us a lot more flexibility in terms of where and how we call these steps. I can see some reusability for when we deploy to a function app with an event grid trigger, for example.

It also gives the user the ability to go back if they accidentally select the wrong pick rather than having to start the entire wizard over.